examples/generate_image_dataset/generate_images.py

```diff
@@ -112,13 +112,27 @@ def single_step(self, action=None, get_observation=False):
 
             # change config based on current scene
             config.defrost()
-            if "kujiale_0000" in pose["scene_id"]:
+
+            if pose["scene_id"] == "kujiale_0000":
                 config.SIMULATION_CONTROLLER.WORLD_PATH_NAME = \
                     "/Game/Scenes/" + pose["scene_id"] + "/Maps/" + pose["scene_id"] + rendering_mode_map_str + "." + pose["scene_id"] + rendering_mode_map_str
-                config.SIMULATION_CONTROLLER.LEVEL_NAME = "/Game/Scenes/" + pose["scene_id"] + "/Maps/" + pose["scene_id"] + rendering_mode_map_str
-            elif "smart_factory_0000" in pose["scene_id"]:
-                config.SIMULATION_CONTROLLER.WORLD_PATH_NAME = "/Game/Scenes/" + pose["scene_id"] + "/Maps/" + pose["scene_id"] + "." + pose["scene_id"]
-                config.SIMULATION_CONTROLLER.LEVEL_NAME = "/Game/Scenes/" + pose["scene_id"] + "/Maps/" + pose["scene_id"]
+                config.SIMULATION_CONTROLLER.LEVEL_NAME = \
+                    "/Game/Scenes/" + pose["scene_id"] + "/Maps/" + pose["scene_id"] + rendering_mode_map_str
+
+                # kujiale_0000 has scene-specific config values
+                scene_config_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), "scene_config.kujiale_0000.yaml")
+
+            elif pose["scene_id"] == "smart_factory_0000":
+                # smart_factory_0000 doesn't need a rendering mode when referring to its map
+                config.SIMULATION_CONTROLLER.WORLD_PATH_NAME = \
+                    "/Game/Scenes/" + pose["scene_id"] + "/Maps/" + pose["scene_id"] + "." + pose["scene_id"]
+                config.SIMULATION_CONTROLLER.LEVEL_NAME = \
+                    "/Game/Scenes/" + pose["scene_id"] + "/Maps/" + pose["scene_id"]
+
+                # smart_factory_0000 has scene-specific config values
+                scene_config_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), "scene_config.smart_factory_0000.yaml")
+
+            config.merge_from_file(scene_config_file)
```

Comment:
Ah, both scenes have scene-specific config values here. I'm even happier with this design than I was before. This is great.